frontend/lib/src/components/widgets/DownloadButton/DownloadButton.tsx

```diff
@@ -63,11 +74,16 @@ function DownloadButton(props: Props): ReactElement {
     kind = BaseButtonKind.TERTIARY
   }
 
+  const downloadUrl = useMemo(
+    () => getDownloadUrl(endpoints, url),
+    [endpoints, url, window.__streamlit?.DOWNLOAD_ASSETS_BASE_URL]
+  )
+
   useEffect(() => {
     // Since we use a hidden link to download, we can't use the onerror event
     // to catch src url load errors. Catch with direct check instead.
-    void endpoints.checkSourceUrlResponse(element.url, "Download Button")
-  }, [element.url, endpoints])
+    void endpoints.checkSourceUrlResponse(downloadUrl, "Download Button")
```

Comment:
I think this fixes a bug in the original implementation independent of the other changes of the PR. The `checkSourceUrlResponse` previously always used `element.url` to make the connection. However, I think it should have been the return value of `endpoints.buildMediaURL(url)` which is the URL that is actually used to download the asset.